window/src/os/windows/window.rs

```diff
@@ -817,6 +893,152 @@ unsafe fn wm_ncdestroy(
+
+        if inner.config.window_decorations != WindowDecorations::RESIZE {
+            return None;
+        }
+
+        // Let the default procedure handle resizing areas
+        let result = DefWindowProcW(hwnd, msg, wparam, lparam);
+
+        if matches!(
+            result,
+            HTNOWHERE
+                | HTRIGHT
+                | HTLEFT
+                | HTTOPLEFT
+                | HTTOP
+                | HTTOPRIGHT
+                | HTBOTTOMRIGHT
+                | HTBOTTOM
+                | HTBOTTOMLEFT
+        ) {
+            return Some(result);
+        }
+
+        // The adjustment in NCCALCSIZE messes with the detection
+        // of the top hit area so manually fixing that.
+        let dpi = GetDpiForWindow(hwnd);
+        let frame_x = GetSystemMetricsForDpi(SM_CXFRAME, dpi) as isize;
+        let frame_y = GetSystemMetricsForDpi(SM_CYFRAME, dpi) as isize;
+        let padding = GetSystemMetricsForDpi(SM_CXPADDEDBORDER, dpi) as isize;
+
+        let coords = mouse_coords(lparam);
+        let cursor_point = screen_to_client(hwnd, ScreenPoint::new(coords.x, coords.y));
+        let is_maximized = window_is_maximized(hwnd);
+
+        // check if mouse is in any of the resize areas (HTTOP, HTBOTTOM, etc)
+
+        let mut client_rect = RECT::default();
+        let client_rect_is_valid =
+            GetClientRect(hwnd, &mut client_rect) == winapi::shared::minwindef::TRUE;
+
+        // Since we are eating the bottom window frame to deal with a Windows 10 bug,
+        // we detect resizing in the window client area as a workaround
+        if !is_maximized
+            && inner.is_win10
+            && client_rect_is_valid
+            && cursor_point.y >= (client_rect.bottom as isize) - (frame_y + padding)
+        {
+            if cursor_point.x <= (frame_x + padding) {
+                return Some(HTBOTTOMLEFT);
+            } else if cursor_point.x >= (client_rect.right as isize) - (frame_x + padding) {
+                return Some(HTBOTTOMRIGHT);
+            } else {
+                return Some(HTBOTTOM);
+            }
+        }
+
+        if !is_maximized && cursor_point.y >= 0 && cursor_point.y < frame_y {
+            if cursor_point.x <= (frame_x + padding) {
+                return Some(HTTOPLEFT);
+            } else if cursor_point.x >= (client_rect.right as isize) - (frame_x + padding) {
+                return Some(HTTOPRIGHT);
+            } else {
+                return Some(HTTOP);
+            }
+        }
+
+        return Some(HTCLIENT);
+    }
+
+    None
+}
+
+fn window_is_maximized(hwnd: HWND) -> bool {
+    let mut placement = WINDOWPLACEMENT {
+        length: std::mem::size_of::<WINDOWPLACEMENT>() as _,
+        ..Default::default()
+    };
+
+    if unsafe { GetWindowPlacement(hwnd, &mut placement) } != winapi::shared::minwindef::TRUE {
+        false
+    } else {
+        placement.showCmd == SW_SHOWMAXIMIZED as u32
+    }
+}
+
+fn is_win10() -> bool {
```

Comment:
To avoid making these calls repeatedly, I think we can cache this in a global and then the rest of the code can consult `IS_WIN10`.

Or alternatively, rename `is_win10` to `is_win10_impl` and then make `is_win10` return `IS_WIN10` if you'd prefer to avoid having lots of shouty case.

```suggestion
lazy_static! {
  static ref IS_WIN10: bool = is_win10();
}  

fn is_win10() -> bool {
```